compiler/ztests/sql/subqueries.yaml

```diff
@@ -11,4 +11,4 @@ outputs:
     data: |
       {v:2}
       // ===
-      {x:{y:2}}
+      {x:2}
```

Comment:
This is surprising.

This is even more surprising:

```console
$ go run ./cmd/super -c 'select (values {y:0,z:1})'
{"(  values {y:0,z:1}\n)":0}
```

Why does the new behavior make sense?

Reply:
The subquery logic needs a bit more work.  I have a branch that is waiting on merging this one...
https://github.com/brimdata/super/blob/fa404ad7f24f40b79c139dbd421dcefcb519c41c/compiler/semantic/expr.go#L1140